### PR TITLE
Add rank to model, handlers and tests

### PR DESCRIPTION
**Pull request by amitch23**
This PR adds the optional rank value to the form1299 model, as well as sets the value in the submission and index handlers. 

Testing Steps:
- [x]  `make server_test` passes
- [x]  Can successfully create form instance with rank value
screen shot 2018-02-08 at 11 32 05 am

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -700,6 +698,7 @@ definitions:
       - updated_at
   ServiceMemberRank:
     type: string
+    x-nullable: true
     enum:
```
Comment (macrael) [inline swagger.yaml]: Can we go ahead and capitalize the non-capitalized enums? I think we should standardize on all enums being IN_ALL_CAPS_WITH_UNDERSCORES

**Comment by amitch23** [comment]
@macrael - if I post with an invalid rank value, the response is: `{
  "code": 606,
  "message": "service_member_rank in body should be one of [E_1 E_2 E_3 E_4 E_5 E_6 E_7 E_8 E_9 W_1 W_2 W_3 W_4 W_5 O_1 O_2 O_3 O_4 O_5 O_6 O_7 O_8 O_9 O_10 academy_cadet academy_graduate aviation_cadet civilian_employee midshipman]"
}`. 

It's not a 400 error, but it looks like go-swagger is validating the value. Is this what you mean?

**Comment by macrael** [comment]
Yes, that's all I expect for now. The HTTP status code (as opposed to the "code" in the body) should be a 400 series error.

File: swagger.yaml (hunk 1 above)
Comment (tinyels) [reply swagger.yaml]: Let's make that another story

Comment (amitch23) [reply swagger.yaml]: Already added!

**Pull request merged**